File: app/src/main/java/com/jimzrt/umsmounter/Tasks/CheckFolderTask.java
package com.jimzrt.umsmounter.Tasks;


import android.os.Environment;

import java.io.File;


public class CheckFolderTask extends BaseTask {

    public CheckFolderTask(){
        this.name = "Check Folder";
        this.description = "Checking Folders...";
    }
    @Override
    public boolean execute() {

        String path = Environment.getExternalStorageDirectory().toString()+"/UMSMounter";
        File f = new File(path);

        if(!f.exists()){
                    f.mkdir();
                    this.result = "Path " + path + " not found, creating.\n";
                }

                File files[] = f.listFiles();
                if(files != null) {

                    this.result = "Found path " + path + " with " + files.length + " files.\n";
                }

        return true;


    }
}

File: app/src/main/java/com/jimzrt/umsmounter/tasks/CheckFolderTask.kt
package com.jimzrt.umsmounter.tasks

import com.jimzrt.umsmounter.activities.MainActivity
import com.jimzrt.umsmounter.utils.Helper
import java.io.File

class CheckFolderTask : BaseTask() {
    override fun execute(): Boolean {
        val path = MainActivity.USERPATH
        var f = File(path)
        var success = true
        if (!f.exists()) {
            success = f.mkdir()
            result = "Path $path not found, creating.\n"
        }
        val cachePath = Helper.combinePaths(MainActivity.USERPATH, MainActivity.CACHEDIR)
        f = File(cachePath)
        if (!f.exists()) {
            success = f.mkdir()
            result = "Cache path $cachePath not found, creating.\n"
        }
        return success
    }

    init {
        name = "Check Folder"
        description = "Checking Folders..."
    }
}